src/ypspur_ros.cpp

```diff
@@ -852,22 +861,30 @@ class YpspurRosNode
           y = odom.pose.pose.position.y + dt * v * sinf(yaw);
         }
 
-        odom.header.stamp = ros::Time(t);
-        odom.pose.pose.position.x = x;
-        odom.pose.pose.position.y = y;
-        odom.pose.pose.position.z = 0;
-        odom.pose.pose.orientation = tf2::toMsg(tf2::Quaternion(z_axis_, yaw));
-        odom.twist.twist.linear.x = v;
-        odom.twist.twist.linear.y = 0;
-        odom.twist.twist.angular.z = w;
-        pubs_["odom"].publish(odom);
-
-        odom_trans.header.stamp = ros::Time(t) + ros::Duration(tf_time_offset_);
-        odom_trans.transform.translation.x = x;
-        odom_trans.transform.translation.y = y;
-        odom_trans.transform.translation.z = 0;
-        odom_trans.transform.rotation = odom.pose.pose.orientation;
-        tf_broadcaster_.sendTransform(odom_trans);
+        const ros::Time current_stamp(t);
+        if (!avoid_publishing_duplicated_odom_ || (current_stamp > previous_odom_stamp_))
```

Comment:
I think `previous_odom_stamp_` should be reset on time jump back.

Reply:
yp-spur resets timestamp estimator when detected time jump larger than 0.5s. (https://github.com/openspur/yp-spur/blob/6dfcbfe2b90c5b929ff40f8c6c18ee60bffb436b/src/odometry.c#L405-L412)

Reply:
Does "reset" mean to set the value to zero?
Currently, `previous_odom_stamp_` is changed to `current_stamp` regardless of which value is greater.
Is not it enough?

Reply:
If the system clock is jumped back, `current_stamp > previous_odom_stamp_` will be always false until the system clock gets over the time before the jump.

Reply:
[L887](https://github.com/nhatao/ypspur_ros/blob/95c10ad10144a1a098e8bdbefd40b3fca767c9df/src/ypspur_ros.cpp#L887)(`previous_odom_stamp_ = current_stamp;`) is out of the if-block publishing odometry, so odometry publishing is skipped only once just after the system clock is jumped back.

Reply:
Ah, I overlooked it.